src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.tsx

```diff
@@ -0,0 +1,303 @@
+  textTransform: 'uppercase',
+  textAlign: 'center',
+}));
+
+const NewsletterMenu = ({ accountListId }: Props): ReactElement<Props> => {
+  const { t } = useTranslation();
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [selectedMenuItem, changeSelectedMenuItem] = useState(-1);
+  const [newsletterMenuDialogOpen, changeNewsletterMenuDialogOpen] = useState(
+    false,
+  );
+  const [emailList, changeEmailList] = useState('');
+
+  const {
+    data: {
+      taskAnalytics: {
+        lastElectronicNewsletterCompletedAt,
+        lastPhysicalNewsletterCompletedAt,
+      } = {},
+    } = {},
+    loading,
+  } = useGetTaskAnalyticsQuery({
+    variables: { accountListId },
+  });
+
+  const [
+    loadEmailNewsletterContacts,
+    { data: newsletterContactsData, loading: contactsLoading },
+  ] = useGetEmailNewsletterContactsLazyQuery({
+    variables: { accountListId },
+  });
+
+  useEffect(() => {
+    if (newsletterContactsData?.contacts.nodes.length > 0) {
+      changeEmailList(
+        newsletterContactsData.contacts.nodes.reduce(
+          (result, contact, index) => {
+            return contact?.primaryPerson?.primaryEmailAddress
+              ? index === 0
+                ? `${contact.primaryPerson.primaryEmailAddress.email}`
+                : `${result},${contact.primaryPerson.primaryEmailAddress.email}`
+              : result;
+          },
+          '',
+        ),
+      );
+    }
+  }, [newsletterContactsData]);
+
+  const handleMenuOpen = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleMenuClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleMenuItemClick = (menuItem) => {
+    changeSelectedMenuItem(menuItem);
+    changeNewsletterMenuDialogOpen(true);
+    switch (menuItem) {
+      case 0:
+        break;
+      case 1:
+        loadEmailNewsletterContacts();
+        break;
+      case 2:
+        break;
+    }
+  };
+
+  const handleDialogClose = () => {
+    changeNewsletterMenuDialogOpen(false);
+    changeSelectedMenuItem(-1);
+  };
+
+  const latestNewsletterDate = () => {
+    const electronicDate = DateTime.fromISO(
+      lastElectronicNewsletterCompletedAt,
+    );
+    const physicalDate = DateTime.fromISO(lastPhysicalNewsletterCompletedAt);
+
+    if (
+      !lastElectronicNewsletterCompletedAt &&
+      !lastPhysicalNewsletterCompletedAt
+    ) {
+      return t('never');
+    } else if (
+      lastElectronicNewsletterCompletedAt &&
+      !lastPhysicalNewsletterCompletedAt
+    ) {
+      return electronicDate.toLocaleString();
+    } else if (
+      !lastElectronicNewsletterCompletedAt &&
+      lastPhysicalNewsletterCompletedAt
+    ) {
+      return physicalDate.toLocaleString();
+    } else {
+      return electronicDate < physicalDate
+        ? physicalDate.toLocaleString()
+        : electronicDate.toLocaleString();
+    }
+  };
+  // TODO: Finish creating dialog for "Log Newsletter" and "Export Physical"
+  const renderDialog = () => {
```

Comment:
Also very important feedback I'd love to hear, how do you feel about how I approached rendering these dialogs? Could it be done better? I didn't want to create a separate component for each of these menu items so this seemed like the best way to handle it. Let me know what you guys think 🙂

Reply:
Instead of several different switches, might be better to group the title, content, action etc into an array of objects... 🤷‍♂️